### PR TITLE
Update docs on kedro integration and deployment

### PR DESCRIPTION
**Pull request by antonymilne**
## Description

A few corrections, clarification and additions to the docs.

## Checklist

- [x] I have not referenced individuals, products or companies in any commits, directly or indirectly
- [x] I have not added data or restricted code in any commits, directly or indirectly
- [x] I have updated the docstring of any public function/class/model changed
- [x] I have added tests to cover my changes (if applicable)

## Types of changes

- [x] Docs/refactoring (non-breaking change which improves codebase)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: vizro-core/docs/pages/user_guides/run.md [hunk 1]
```diff
@@ -59,9 +59,17 @@ The dashboard application can be launched in jupyter environment in `inline`, `e
 - you can specify `jupyter_mode="external"` and a link will be displayed to direct you to the localhost where the dashboard is running.
 - you can use tab mode by `jupyter_mode="tab"` to automatically open the app in a new browser
 
-## Launch it via Gunicorn
+## Deployment of Vizro app
```
Comment (lingyielia) [inline vizro-core/docs/pages/user_guides/run.md:62]: To be very accurate, this instruction is still more focus on "running the app with gunicorn" than how to deploy the app 🤔 

Shall we phrase it as something like, for deployment, it's recommended to run the app with gunicorn, and here is how

**Review by lingyielia — APPROVED** [review]
LGTM!

File: vizro-core/docs/pages/user_guides/run.md (hunk 1 above)
Comment (maxschulz-COL) [reply vizro-core/docs/pages/user_guides/run.md:62]: Agreed, in the current form, this point breaks the reading flow. This entire guide has the overarching title of "How to launch the dashboard".

I was going to suggest to write a separate section on how to deploy the dashboard, when I realised this is exactly what we had 😅 .

I would suggest keeping the old structure, but to maybe enhance it with the additional links and so on.

Overall then I think the title of this section should change back to "Gunicorn", and ideally the one above to simply "Jupyter Environment". Then the sections alrady contain the answer to the user guide question, which I think is an improvement.

**Review by maxschulz-COL — CHANGES_REQUESTED** [review]
Lgtm! Thanks for improving the guide on kedro integrations! Let's discuss the changes on `run.md` where I think we need to consider the overall reading flow a little

File: vizro-core/docs/pages/user_guides/run.md (hunk 1 above)
Comment (antonymilne) [reply vizro-core/docs/pages/user_guides/run.md]: See what you think of the new version 🙂 I even did the special popup thing:
image

File: vizro-core/docs/pages/user_guides/run.md [hunk 2]
```diff
@@ -80,24 +81,22 @@ The dashboard application can be launched in jupyter environment in `inline`, `e
 
         dashboard = vm.Dashboard(pages=[page])
         app = Vizro().build(dashboard)
-        server = app.dash.server
+        server = app.dash.server # (1)!
         ```
-You need to expose the server via `app.dash.server` in order to use gunicorn as your wsgi here.
-Run it via
+
+        1.  Expose the underlying Flask app through `app.dash.server`.
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/run.md:87]: ❤️

**Review by maxschulz-COL — APPROVED** [review]
Like it! Thanks for applying the changes :)

**Pull request merged**